app/models/article.rb

```diff
@@ -115,6 +120,38 @@ def increment_view_count
 
   private
 
+  def category_id_changed_action
+    # We need to update the position of the article in the new category
+    return unless persisted?
+
+    # this means the article is just created
+    # and the category_id is newly set
+    # and the position is already present
+    return if created_at_before_last_save.nil? && position.present? && category_id_before_last_save.nil?
```

Comment:
@tejaswinichile @sojan-official need your thoughts on this logic.. This is to ensure the category ID is not overridden on create, if the position is supplied when creating the article

Reply:
This seems okay.